Math/NumberTheory/ArithmeticFunctions/Standard.hs

```diff
@@ -0,0 +1,300 @@
+-- |
+-- Module:      Math.NumberTheory.ArithmeticFunctions.Standard
+-- Copyright:   (c) 2016 Andrew Lelechenko
+-- Licence:     MIT
+-- Maintainer:  Andrew Lelechenko <andrew.lelechenko@gmail.com>
+-- Stability:   Provisional
+-- Portability: Non-portable (GHC extensions)
+--
+-- Textbook arithmetic functions.
+--
+
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE ViewPatterns        #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Math.NumberTheory.ArithmeticFunctions.Standard
+  ( -- * Multiplicative functions
+    multiplicative
+  , divisors, divisorsA
+  , divisorsSmall, divisorsSmallA
+  , tau, tauA
+  , sigma, sigmaA
+  , totient, totientA
+  , jordan, jordanA
+  , moebius, moebiusA
+  , liouville, liouvilleA
+    -- * Additive functions
+  , additive
+  , smallOmega, smallOmegaA
+  , bigOmega, bigOmegaA
+    -- * Misc
+  , carmichael, carmichaelA
+  , expMangoldt, expMangoldtA
+  ) where
+
+import Data.IntSet (IntSet)
+import qualified Data.IntSet as IS
+import Data.Set (Set)
+import qualified Data.Set as S
+import Data.Semigroup
+
+import Unsafe.Coerce
+
+import Math.NumberTheory.ArithmeticFunctions.Class
+import Math.NumberTheory.UniqueFactorisation
+
+import Numeric.Natural
+
+#if MIN_VERSION_base(4,8,0)
+#else
+import Data.Foldable
+import Data.Word
+#endif
+
+#if MIN_VERSION_base(4,7,0)
+import Data.Coerce
+#else
+coerce :: a -> b
+coerce = unsafeCoerce
+#endif
+
+-- There are special rewrite rules for (^ Int), but not for (^ Word).
```

Comment:
fromIntegral has a good specialization rule for this conversion, I think, as does Data.Bits, perhaps one of those would be what you want?